src/pepr/operator/controllers/keycloak/authservice/config.ts

```diff
@@ -1,29 +1,48 @@
 import { createHash } from "crypto";
-
 import { K8s, kind } from "pepr";
 import { UDSConfig } from "../../../../config";
 import { Client } from "../types";
 import { buildChain, log } from "./authservice";
 import { Action, AuthserviceConfig } from "./types";
 
+// Cache for in-memory secret to avoid unnecessary Kubernetes secret lookups
+let inMemorySecret: AuthserviceConfig | null = null;
+
+// Track pending package updates and their resolve functions
+const pendingPackages: Map<AuthserviceConfig, () => void> = new Map();
+
+// Backup for the last known successful state of the secret
+let lastSuccessfulSecret: AuthserviceConfig | null = null;
+
+// Timer for debouncing updates to the secret
+let debounceTimer: NodeJS.Timeout | null = null;
+
+// Debounce duration (12 seconds) to reduce excessive updates, configurable via environment variable
+const DEBOUNCE_DURATION = parseInt(process.env.DEBONCE_DURATION || "12000", 10);
```

Comment:
IMO this should be way lower.  We need to balance the responsiveness of reconciling a single package.  

I would probably test by applying `n` packages at the same time.  If those packages are updating as part of the same secret update, lower the debounce time.  Repeat until the logs indicate there are multiple updates happening and then dial it back.

Reply:
@rjferguson21 yeah good call out. I came to this number because i was testing with 8 packages and this is how long the debounce needed for it to be a single update vs two updates. Not sure what the best number of packages would be to use maybe half that? 4-5